DASH-IF-IOP.bs.md

```diff
@@ -10,8 +10,500 @@ Live discussion in #document-authoring on Slack.
+
+Note: In addition to mapping the content to real time, [[#timing-mpd-updates|a dynamic MPD can be updated during the presentation]]. Updates may add new periods and remove or modify existing ones.
+
+The zero point in the MPD timeline of a dynamic MPD SHALL be mapped to the point in real time indicated by `MPD@availabilityStartTime`. This value SHALL NOT change between MPD updates.
+
+The ultimate purpose of the MPD is to enable the client to obtain media samples for playback. The following data structures are most relevant to locating and scheduling the samples:
+
+1. The MPD consists of consecutive [=periods=] which map data onto the MPD timeline.
+1. Each period contains of one or more [=representations=], each of which provides media samples inside a sequence of [=media segments=].
+1. Representations within a period are grouped in [=adaptation sets=], which associate related representations and decorate them with metadata.
+
+<figure>
+	<img src="Images/Timing/BasicMpdElements.png" />
+	<figcaption>The primary elements described by an MPD.</figcaption>
+</figure>
+
+The chapters below explore these relationships in detail.
+
+## Periods ## {#timing-period}
+
+An MPD SHALL define an ordered list of one or more <dfn title="period">periods</dfn>. A period is both a time span on the [=MPD timeline=] and a definition of the data to be presented during the period. Period timing is relative to the zero point of the [=MPD timeline=].
+
+<figure>
+	<img src="Images/Timing/PeriodsMakeTheMpd.png" />
+	<figcaption>An MPD is a collection of periods, starting from a zero point and with a total duration (which may be unlimited for live services).</figcaption>
+</figure>
+
+Common reasons for defining multiple periods are:
+
+* Assembling a presentation from multiple self-contained pieces of content.
+* Inserting ads in the middle of existing content and/or replacing spans of existing content with ads.
+* Adding/removing certain representations as the nature of the content changes (e.g. a new title starts with a different set of offered languages).
+* Updating period-scoped metadata (e.g. codec configuration or DRM signaling).
+
+Perioods are self-contained - a service SHALL NOT require a client to know the contents of another period in order to correctly present a period. Knowledge of the contents of different periods MAY be used by a client to achieve seamless period transitions, especially when working with [[#timing-connectivity|period-connected representations]].
+
+All periods SHALL be consecutive and non-overlapping. A period MAY have a duration of zero.
+
+Note: A period with a duration of zero might, for example, be the result of ad-insertion logic deciding not to insert any ad.
```

Comment:
Possibly in some cases it could omit them, indeed. However, I copied it over as-is from v4 (which allows it) intentionally, as it seems reasonably harmless while also being likely implemented this way by existing services who would be reticent to change.

Do you see an impactful benefit from forbidding zero-length periods?

Reply:
I think clients can handle this case pretty trivially. I also think servers can pretty trivially remove them. So from the perspective that clients should have to handle as few cases as possible (and in particular edge cases), that would point toward having them removed on the server side. That said, it's probably not a big deal to leave this as-is, if existing services do this already.

Reply:
To add to this, xlink periods are resolved client-side, so the client will need to be zero-duration period aware anyway if it wants to support xlink.